src/job_cache/daemon_cache.cpp

```diff
@@ -1001,6 +1004,11 @@ void DaemonCache::add(const AddJobRequest &add_request) {
   std::string tmp_job_dir = "tmp_" + rng.unique_name();
   mkdir_no_fail(tmp_job_dir.c_str());
 
+  // Get the time so we know when this job was created
+  timespec tp;
+  clock_gettime(CLOCK_REALTIME, &tp);
+  int64_t time = 1000000ll * int64_t(tp.tv_sec) + tp.tv_nsec / 1000;
```

Comment:
nit is there not some magic "get unix epoc" function defined somewhere?

Reply:
Its also worth considering SQLite3's datatime support for this https://www.sqlite.org/datatype3.html

Reply:
This is the easiest way I'm aware of to get microseconds

Reply:
is per-second resolution good enough?

Reply:
No its not. It makes testing for LRU stuff super hard and makes fuzz testing for eviction annoying as well because you have to sleep a whole second per in order to get the evictions to happen in the right order. But at the microsecond level everything just works because it takes around 100x to 1000x that long to do  a cache read.